app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx

```diff
@@ -0,0 +1,227 @@
+          <PathText path={path} availableWidth={50} />
+          <div className="file-conflicts-status">No conflicts remaining</div>
+        </div>
+        <div className="green-circle">
+          <Octicon symbol={OcticonSymbol.check} />
+        </div>
+      </li>
+    )
+  }
+
+  private renderConflictedFile(
+    path: string,
+    conflicts: number,
+    editorName: string | undefined,
+    onOpenEditorClick: () => void
+  ): JSX.Element {
+    const humanReadableConflicts = Math.ceil(conflicts / 3)
+    const message =
+      humanReadableConflicts === 1
+        ? `1 conflict`
+        : `${humanReadableConflicts} conflicts`
+    return (
+      <li className="unmerged-file-status-conflicts">
+        <Octicon symbol={OcticonSymbol.fileCode} className="file-octicon" />
+        <div className="column-left">
+          <PathText path={path} availableWidth={50} />
+          <div className="file-conflicts-status">{message}</div>
+        </div>
+        <Button onClick={onOpenEditorClick}>
+          {this.editorButtonString(editorName)}
+        </Button>
+      </li>
+    )
+  }
+
+  private renderUnmergedFile(
+    file: WorkingDirectoryFileChange,
+    editorName: string | undefined,
+    repositoryPath: string
+  ): JSX.Element | null {
+    switch (file.status) {
+      case AppFileStatus.Resolved:
+        return this.renderResolvedFile(file.path)
+      case AppFileStatus.Conflicted:
+        return this.renderConflictedFile(
+          file.path,
+          file.conflictMarkers,
+          editorName,
+          () =>
+            this.props.openFileInExternalEditor(join(repositoryPath, file.path))
+        )
+      default:
+        return null
+    }
+  }
+
+  private renderUnmergedFiles(
+    files: Array<WorkingDirectoryFileChange>,
+    editorName: string | undefined,
+    repositoryPath: string
+  ) {
+    return (
+      <ul className="unmerged-file-statuses">
+        {files.map(f => this.renderUnmergedFile(f, editorName, repositoryPath))}
+      </ul>
+    )
+  }
+
+  private getUnmergedFiles() {
+    return this.props.status.files.filter(
+      file =>
+        file.status === AppFileStatus.Conflicted ||
+        file.status === AppFileStatus.Resolved
+    )
+  }
+
+  private renderUnmergedFilesSummary(unmergedFiles: number) {
+    // localization, it burns :vampire:
+    const message =
+      unmergedFiles === 1
+        ? `1 conflicted file`
+        : `${unmergedFiles} conflicted files`
+    return <h3 className="summary">{message}</h3>
+  }
+
+  public render() {
+    const unmergedFiles = this.getUnmergedFiles()
+    const anyConflictedFiles = unmergedFiles.some(
+      f => f.status === AppFileStatus.Conflicted
+    )
+    const titleString = this.titleString(
+      this.props.currentBranch.name,
+      this.props.comparisonBranch.name
+    )
+    const tooltipString = anyConflictedFiles
+      ? 'Resolve all changes before merging'
+      : undefined
+    const openThisRepositoryInShell = () =>
```

Comment:
Rather than creating this on each `render`, can we lift this up to be an instance function that only needs to be created once?

```tsx
  private openThisRepositoryInShell = () => {
    this.props.openRepositoryInShell(this.props.repository)
  }

  private renderShellLink(): JSX.Element {
    return (
      <div className="cli-link">
        You can also{' '}
        <a onClick={this.openThisRepositoryInShell}>open the command line</a> to
        resolve
      </div>
    )
  }
```